libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -389,6 +400,50 @@ void AP_Scheduler::Log_Write_Performance()
     AP::logger().WriteCriticalBlock(&pkt, sizeof(pkt));
 }
 
+// display task statistics as text buffer for @SYS/tasks.txt
+size_t AP_Scheduler::task_info(char *buf, size_t bufsize) const
+{
+    size_t total = 0;
+
+    // a header to allow for machine parsers to determine format
+    int n = snprintf(buf, bufsize, "TasksV1\n");
+    if (n <= 0 || perf_info.get_task_info(0) == nullptr) {
+        return 0;
+    }
+    buf += n;
+    bufsize -= n;
+    total += n;
+
+    uint8_t i = 0;
+
+    do {
+        const AP_Scheduler::Task& task = (i < _num_unshared_tasks) ? _tasks[i] : _common_tasks[i - _num_unshared_tasks];
+        const AP::PerfInfo::TaskInfo* ti = perf_info.get_task_info(i);
+
+        uint16_t avg = 0;
+        if (ti->tick_count > 0) {
+            avg = ti->elapsed_time_us / ti->tick_count;
+        }
+
+#if HAL_MINIMIZE_FEATURES
+        n = snprintf(buf, bufsize, "%-16.16s MIN=%u MAX=%u AVG=%u OVR=%u SLP=%u\n", task.name,
```

Comment:
I think the TSV format numbers are fine.
The item name should be in the first line only.

Reply:
I'm copying what has been done for the threads.txt, I think we should be consistent in the format.